components/net/http_cache.rs

```diff
@@ -509,11 +553,13 @@ fn handle_range_request(
 impl HttpCache {
     /// Constructing Responses from Caches.
     /// <https://tools.ietf.org/html/rfc7234#section-4>
-    pub fn construct_response(
+    pub(crate) fn construct_response(
```

Comment:
I think this can be made into a function following these changes, which would make it much clearer when this is called while a guard on an entry is held.